src/citationnet/templates/visDynamic.html

```diff
@@ -0,0 +1,117 @@
+        <a class="list-group-item" href="javascript:void(0)" id="btnFORChart" onclick="toggleFORChart()">Fields of Research</a>
+      </ul>
+    </div>
+    <div class="my-2">
+      <h6>Node options</h6>
+      <ul class="list-group">
+        <a class="list-group-item" href="javascript:void(0)" id="btnNodeSize" onclick="window.net.toggleNodeSize()">Size by number of citations</a>
+        <a class="list-group-item" type="button" data-bs-toggle="collapse" data-bs-target="#relnodesize" aria-expanded="false" aria-controls="relnodesize">Relative node size</a>
+      </ul>
+      <div class="collapse input-group my-2" id="relnodesize">
+        <input type="range" min="0.1" max="1.0" value="0.5" step="0.1" class="slider" id="rngNodeSize">
+        <button class="btn btn-primary" onclick="window.net.readNodeSize()">set</button>
+      </div>
+    </div>
+    <div class="mb-2">
+      <h6>Edge options</h6>
+      <ul class="list-group">
+        <a class="list-group-item" href="javascript:void(0)" id="btnEdgesOnlyInput" onclick="window.net.toggleEdgesOnlyInput()">Edges of input only</a>
+      </ul>
+    </div>
+    <div>
+      <h6>Layout options</h6>
+      <ul class="list-group">
+        <a class="list-group-item" type="button" data-bs-toggle="collapse" data-bs-target="#cylinderradius" aria-expanded="false" aria-controls="cylinderradius">Cylinder radius</a>
+      <a class="list-group-item" type="button" data-bs-toggle="collapse" data-bs-target="#outerringvalue" aria-expanded="false" aria-controls="outerringvalue">Outer value</a>
+      <a class="list-group-item" href="javascript:void(0)" id="btnDistanceFromInputNode" onclick="window.net.toggleDistanceFromInputNode()"
+        style="font-weight: bold;">Spacing around input</a>
+      </ul>
+      <div class="collapse input-group my-2" id="cylinderradius">
+        <input type="range" min="0" max="500" value="350" class="slider" id="rngLayoutRadius">
+        <button class="btn btn-primary" onclick="window.net.readLayout()">set</button>
+      </div>
+      <div class="collapse input-group my-2" id="outerringvalue">
+        <input type="range" min="0" max="1000" value="800" class="slider" id="rngLayoutOuterValue">
+        <button class="btn btn-primary" onclick="window.net.readLayout()">set</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="module">
+  import { CitationNet } from "{{ url_for('static', filename='force-citationNet.js') }}";
+
+  const jsondata = await '{{ jsondata|tojson }}';
+  window.net = new CitationNet(jsondata);
+  window.net.initialize(false);
+  document.getElementById("sidebar").style.width = "300px";
+
+  window.adaptWindowSize = () => {
+    var navHeight = document.getElementsByClassName("navbar")[0].scrollHeight;
+
+    window.net.adaptWindowSize();
+    document.getElementById("3d-graph").style.top = navHeight;
+    document.getElementById("sidebar").style.top = navHeight;
+  };
+
+  window.onresize = window.adaptWindowSize;
+
+  await showFOR();
```

Comment:
Instead of relying on `window`, pass `net` to this function to make it accessible from `loadFOR` in pieChart_vega-lite.js